### PR TITLE
add runtime support to dataReconciliation module

### PR DESCRIPTION
**Pull request by arun3688**
**Purpose:**
   Added support for calculating DataReconclilation algorithm. 

**Changes**:
Added a new runtime module DataReconciliation.cpp 
Added support for code-generation to dataReconciliation procedure which generates codes for 
1. Calculating the jacobians named "initialAnalyticJacobianF
2. Calculating SetC- C(x,y) named "setc_function()"

Added a new runtime flags for calculating the DataReconciliation procedure namely
1.  reconcile -- Entry point for DataReconciliation module
2.  sx -- Csv file which specifies the covariance matrix for the dataReconciliation
3.  eps  -- convergence value

### CODE REVIEW

File: SimulationRuntime/c/util/simulation_options.h [hunk 1]
```diff
@@ -144,6 +144,9 @@ enum _FLAG
   FLAG_STEADY_STATE_TOL,
   FLAG_UP_HESSIAN,
   FLAG_W,
+  FLAG_DATA_RECONCILE,
+  FLAG_DATA_RECONCILE_Sx,
+  FLAG_DATA_RECONCILE_Eps,
```
Comment (lochel) [inline SimulationRuntime/c/util/simulation_options.h]: The flags should be sorted in alphabetic order.

Comment (arun3688) [reply SimulationRuntime/c/util/simulation_options.h]: @lochel , added the Flags in alphabetic order see https://github.com/OpenModelica/OMCompiler/pull/2845/commits/8af3b0933f715f6551edc2468f0559a9fd524667

Comment (arun3688) [reply SimulationRuntime/c/util/simulation_options.h]: @lochel sorted the flags according to name

**Comment by OpenModelica-Hudson** [comment]
The job failed the coding style; for details, see [OpenModelica_TEST_PULL_REQUEST 2019-01-04_11-30-12](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6300/console). Remember to clean up the commit you did that breaks the coding style rather than piling a new commit on top of the old ones. This goes double for accidental addition of files that you need to remove.

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2019-01-05_15-32-59](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6302/console).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2019-01-07_13-37-49](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6305/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2019-01-08_16-00-43](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6313/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2019-01-08_16-25-36](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6314/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2019-01-10_11-16-58](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6328/).

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2019-01-10_12-01-58](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6329/console).

**Pull request closed without merging**